modules/_labs/select/react/lib/SelectMenu.tsx

```diff
@@ -20,7 +48,18 @@ interface SelectMenuProps
   /**
    * The ref to the underlying menu/listbox element. Use this to imperatively manipulate the menu.
    */
-  menuRef?: React.Ref<HTMLUListElement>;
+  menuRef?: React.RefObject<HTMLUListElement>;
+}
+
+interface SelectMenuState {
+  // We need the width state to support the use case where the menu is
+  // displayed without user interaction (e.g., if we want to show the menu
+  // from the start). buttonRef.current is set to null during the first
+  // render, which prevents Popper from rendering due to the lack of a
+  // reference element. By the time componentDidMount runs, buttonRef.current
+  // will have been set -- we can then update the width state and trigger
+  // a render with the Popper menu.
+  width: number;
```

Comment:
This is an example of where hooks are easier and don't require state. `useEffect` and `useLayoutEffect` are run after render where an element's dimensions can be measured. `useLayoutEffect` will trigger before DOM changes are rendered to the page and `useEffect` will trigger after DOM changes are rendered to the page.

Reply:
SelectMenu has been converted to a function component. State is now maintained using the `useState` hook.